services/pcs/tests_pcs/test_republish.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Testing for the republish functionality."""
+
+import pytest
+from ghga_event_schemas.pydantic_ import FileDeletionRequested
+from hexkit.correlation import get_correlation_id, set_new_correlation_id
+from hexkit.protocols.daosub import DaoSubscriberProtocol
+from hexkit.providers.akafka import KafkaOutboxSubscriber
+from hexkit.providers.akafka.testutils import ExpectedEvent
+from hexkit.providers.mongokafka.provider import dto_to_document
+
+from pcs.config import Config
+from tests_pcs.fixtures.joint import JointFixture
+
+pytestmark = pytest.mark.asyncio()
+
+
+class DummySubTranslator(DaoSubscriberProtocol):
+    """A class that consumes FileDeletionRequested events."""
+
+    event_topic: str = ""
+    dto_model = FileDeletionRequested
+    correlation_ids: list[str]
+    consumed_events: list[str]
```

Comment:
I would use a single list with two-tuples here.

Reply:
Nice, I have done that in [e510694](https://github.com/ghga-de/file-services-backend/pull/31/commits/e5106941e6db6cfcfb129857acef794ff9353431)